framework/web/Request.php

```diff
@@ -1477,6 +1493,181 @@ protected function loadCookies()
         return $cookies;
     }
 
+    /**
+     * Returns uploaded files for this request.
+     * Uploaded files are returned in format according to [PSR-7 Uploaded Files specs](http://www.php-fig.org/psr/psr-7/#16-uploaded-files).
+     * @return array uploaded files.
+     * @since 2.1.0
+     */
+    public function getUploadedFiles()
+    {
+        if ($this->_uploadedFiles === null) {
+            $this->getBodyParams(); // uploaded files are the part of the body and may be set while its parsing
+            if ($this->_uploadedFiles === null) {
+                $this->_uploadedFiles = $this->defaultUploadedFiles();
+            }
+        }
+        return $this->_uploadedFiles;
+    }
+
+    /**
+     * Sets uploaded files for this request.
+     * Data structure for the uploaded files should follow [PSR-7 Uploaded Files specs](http://www.php-fig.org/psr/psr-7/#16-uploaded-files).
+     * @param array|null $uploadedFiles uploaded files.
+     * @since 2.1.0
+     */
+    public function setUploadedFiles($uploadedFiles)
+    {
+        $this->_uploadedFiles = $uploadedFiles;
+    }
+
+    /**
+     * Initializes default uploaded files data structure parsing super-global $_FILES.
+     * @see http://www.php-fig.org/psr/psr-7/#16-uploaded-files
+     * @return array uploaded files.
+     * @since 2.1.0
+     */
+    protected function defaultUploadedFiles()
+    {
+        $files = [];
+        if (isset($_FILES) && is_array($_FILES)) {
```

Comment:
Are there cases when `$_FILES` isn't available and isn't array?

Reply:
[Nope](http://php.net/manual/en/language.variables.superglobals.php). Good catch. :)